sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/spanner/changestreams/dao/PartitionMetadataAdminDao.java

```diff
@@ -129,7 +123,7 @@ public void createPartitionMetadataTable() {
       // Literals need be added around literals to preserve casing.
       ddl.add(
           "CREATE TABLE \""
-              + tableName
+              + names.getTableName()
```

Comment:
does this method fail if the tables/indexes already exist?

IIUC the issue correctly this is failing currently because the index has the same name across pipelines and the creation is failing.  Do we also have to worry about a single pipeline failing if it created the tables but Dataflow retries the work? It seems we might want ot use create if it doesn't exist.

Reply:
Good point. Since the names have a random component, I assumed the job would fail and we would re-generate them.

Updated with IF NOT EXISTS to circumvent this problem.